### PR TITLE
[Merged by Bors] - feat(PathConnected): review API, add missing lemmas

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Connected/PathConnected.lean [hunk 1]
```diff
@@ -1120,66 +1113,72 @@ class LocPathConnectedSpace (X : Type*) [TopologicalSpace X] : Prop where
...
+  path_connected_basis x := by
+    rw [hasBasis_self]
+    intro t ht
+    rcases (h x).mem_iff.mp ht with ⟨i, hpi, hi⟩
+    exact ⟨s x i, (h x).mem_of_mem hpi, h' x i hpi, hi⟩
+
+@[deprecated (since := "2024-10-16")]
+alias locPathConnected_of_bases := LocPathConnectedSpace.of_bases
+
+section LocPathConnectedSpace
```
Comment (grunweg) [inline Mathlib/Topology/Connected/PathConnected.lean:1128]: Should this `section` go above the previous declaration also?

**Review by grunweg — COMMENTED** [review]
This is a nice golf, and I agree the new proofs are generally better than the old ones. Thanks!
maintainer merge

**Comment by grunweg** [comment]
Please address the deprecation warnings, though.

**Comment by sgouezel** [comment]
bors r+
Thanks!

**Pull request closed without merging**